include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -227,16 +229,22 @@ struct Value : Base<OrtValue> {
   TensorTypeAndShapeInfo GetTensorTypeAndShapeInfo() const;
 };
 
-struct Allocator : Base<OrtAllocator> {
-  static Allocator CreateDefault();
+struct Allocator {
```

Comment:
Had to change this because Base relies on the existence of OrtRelease function which we don't want to support for this statically allocated allocator.

Reply:
Is there any other use for this class besides holding the Allocator with default options?

Reply:
But this change will affect all the alloactors, right? Not just the default one.

Reply:
Currently this seems to be used only for allocators with default options. One idea is to rename this class to AllocatorWithDefaultOptions. That way if we've a different use, we can have a generic Allocator class that (derives from Base) and hence supports creation and deletion every single time. Would renaming it to AllocatorWithDefaultOptions make it more clear?

Reply:
@RyanUnderhill and I had an offline conversation and we think that using a new name AllocatorWithDefaultOptions + creating it in the constructor might be the lesser of the evils for this. Updated code.